Sources/NIOHTTP2/MultiplexerAbstractChannel.swift

```diff
@@ -89,24 +90,36 @@ extension MultiplexerAbstractChannel {
         }
     }
 
-    func configureInboundStream(initializer: InboundStreamStateInitializer) {
+    func configureInboundStream(initializer: InboundStreamStateInitializer, promise: EventLoopPromise<Any>?) {
         switch initializer {
         case .includesStreamID(let initializer):
             self.baseChannel.configure(initializer: initializer, userPromise: nil)
         case .excludesStreamID(let initializer):
-            self.baseChannel.configure(initializer: initializer, userPromise: nil)
+            let channelPromise: EventLoopPromise<Channel>?
+            if let promise = promise {
+                channelPromise = self.baseChannel.eventLoop.makePromise(of: Channel.self)
+                channelPromise!.completeWith(promise.futureResult.map { value in value as! Channel })
+            } else {
+                channelPromise = nil
+            }
+            self.baseChannel.configure(initializer: initializer, userPromise: channelPromise)
+        case .returnsAny(let initializer):
+            self.baseChannel.configure(initializer: initializer, userPromise: promise)
```

Comment:
I am a bit confused here but might just lack the context. Are we using both cases because one is the inbound stream and the other is creating a new outbound stream case?

Reply:
Sorry, I don't follow the question. Both cases of what? I'll take a punt at clarifying some things.

For initializer type:
- `.includesStreamID` is the legacy case
- `.excludesStreamID` is the non-async case
- `.returnsAny` is the async case

We optionally remap the promise type if we have one from `Any` to `Channel` but only if we have one. In the past we didn't supply a promise on the inbound path so if we aren't in async world then we don't supply one now.

Reply:
The thing I am confused then is why did something in the implementation for `.excludesStreamID` change now. If this is the non-async case can we actually hit this with a non-nil promise?